docs/4.2. Behaviour - Clients.md

```diff
@@ -20,13 +20,26 @@ containing the required details).
 
 Clients are RECOMMENDED to provide for the OAuth 2.0 Dynamic Client Registration Protocol [RFC 7591][RFC-7591] to allow
 for zero-configuration setup. NMOS Nodes SHOULD be capable of registering dynamically in order to limit the
-configuration associated with large numbers of Nodes coming online simultaneously. Clients MUST include a populated
-`scope` parameter when performing dynamic registrations.
+configuration associated with large numbers of Nodes coming online simultaneously. Clients MUST include populated
+`client_name` and `scope` parameters when performing dynamic registrations. The `client_name` MUST be unique to each
+instance of the software or hardware. It is RECOMMENDED that the `client_name` include the manufacturer's name, product
+identifier and serial number.
 
 Clients that do not register with an explicit `token_endpoint_auth_method` parameter will be registered as a
 Confidential Client and issued with a client secret. Public clients MUST specify a value of "none" for this parameter
 to notify the Authorization Server that is is a public client.
 
+Clients which require use of the `client_credentials` grant type MUST support the following means of authentication
+during the dynamic client registration process. All other Clients SHOULD support these authentication methods:
+*   Presentation of a software statement as defined in Section 2.3 of [RFC 7591][RFC-7591]
+*   Presentation of an initial JSON Web Token (JWT) as suggested in Section 3 of [RFC 7591][RFC-7591]
+*   Unauthenticated registration
+
+Clients MUST be configurable to choose between the above authentication methods, which are listed in order of
+preference. Public keys which can be used to verify software statement signatures MUST be made available for end users
```

Comment:
Is this plug and play enough?

Reply:
The intent was to make sure that the software statement is the default as this is the 'most plug and play' method. That's only indicated by it being top of the order of preference though, so if it would benefit from being more explicit I can adjust.

Reply:
The main concern was installing public keys for each device in the auth server, but this is equivalent to the EST setup, so it's each vendor so it's only the first device that may need human configuration